src/components/map/hooks/use-map-interaction.tsx

```diff
@@ -0,0 +1,107 @@
+import mapboxgl from "mapbox-gl";
+import { useEffect, useRef, useState } from "react";
+import { useUrlState } from "../../router/store";
+import { useMapConstants } from "./use-map-constants";
+
+export function useMapInteraction(map: mapboxgl.Map | undefined) {
+  const { ZOOMED_IN_ZOOM_LEVEL } = useMapConstants();
+
+  const setSearchParams = useUrlState((state) => state.setSearchParams);
+
+  const [hoveredTreeId, setHoveredTreeId] = useState<string | undefined>(
+    undefined,
+  );
+  const hoveredTreeIdRef = useRef<string | undefined>(undefined);
+  useEffect(() => {
+    hoveredTreeIdRef.current = hoveredTreeId;
+  }, [hoveredTreeId]);
+
+  const [selectedTreeId, setSelectedTreeId] = useState<string | undefined>(
+    undefined,
+  );
+  const selectedTreeIdRef = useRef<string | undefined>(undefined);
+  useEffect(() => {
+    selectedTreeIdRef.current = selectedTreeId;
+  }, [selectedTreeId]);
```

Comment:
suggestion/comment: I find using plain `useEffect` always a bit hard to read/understand. What do you think about extracting line 19-25 into a named function `useSelectedTreeId` which just returns the ref? Same goes for the `hoveredTreeId` line 11-17

Reply:
good idea. done